frontend/src/App.tsx

```diff
@@ -37,17 +40,24 @@ import PluginList from 'pages/plugins/PluginList';
 import CreatePlugin from 'pages/plugins/create/CreatePlugin';
 import Projects from 'pages/projects/Projects';
 import ProjectDetail from 'pages/projects/detail/ProjectDetail';
-import React, { Suspense, useEffect, useState } from 'react';
+import UserList from 'pages/user/UserList';
+import React, { Suspense, useContext, useEffect, useState } from 'react';
 import { useTranslation } from 'react-i18next';
 import { AuthProvider, AuthProviderProps, useAuth } from 'react-oidc-context';
 import { Route, BrowserRouter as Router, Routes } from 'react-router-dom';
-import { CONFIG_URL, PROJECT_CONFIG_JSON } from 'ts/const';
+import { CONFIG_URL, IUserRole, PROJECT_CONFIG_JSON } from 'ts/const';
 import Home from './pages/home/Home';
 
 const LoginCallback: React.FC = () => {
+  const currentUser = useContext(UserContext);
+
   useEffect(() => {
     const baseUrl = '/';
-    window.location.href = baseUrl;
+    if (currentUser?.role === IUserRole.ANALYST) {
```

Comment:
If user refresh the page (F5), the page will always redirect `/` or `/analytics`. 
For example: When user in pipeline detail page. they want to refresh the data and refresh...

Reply:
The `useEffect` in **LoginCallback**, so it has no impact on refresh.